src/plugins/support/PMRSupport/src/pmrwebservice.cpp

```diff
@@ -0,0 +1,544 @@
+static const char *NameProperty       = "Name";
+
+//==============================================================================
+
+void PmrWebService::sendPmrRequest(const PmrRequest &pPmrRequest,
+                                   const QString &pUrl,
+                                   const Action pAction,
+                                   const QString &pName)
+{
+    // Let the user know that we are busy
+
+    emit busy(true);
+
+    // Send our request to the PMR, asking for the response to be compressed,
+    // but only if we are connected to the Internet
+
+    if (Core::internetConnectionAvailable()) {
+        QNetworkRequest networkRequest;
+
+        networkRequest.setRawHeader("Accept", "application/vnd.physiome.pmr2.json.1");
+        networkRequest.setRawHeader("Accept-Encoding", "gzip");
+
+        switch (pPmrRequest) {
+        case ExposuresList:
+            networkRequest.setUrl(QUrl("https://models.physiomeproject.org/exposure"));
+
+            break;
+        case ExposureInformation:
+        case WorkspaceInformation:
+        case ExposureFileInformation:
+            networkRequest.setUrl(QUrl(pUrl));
+
+            break;
+        }
+
+        QNetworkReply *networkReply = mNetworkAccessManager->get(networkRequest);
+
+        // Keep track of the type of our PMR request and of the action to carry
+        // out (once we have retrieved everything we need from PMR)
+
+        networkReply->setProperty(PmrRequestProperty, pPmrRequest);
+        networkReply->setProperty(ActionProperty, pAction);
+        networkReply->setProperty(NameProperty, pName);
+    } else {
+        finished();
+    }
+}
+
+//==============================================================================
+
+void PmrWebService::doCloneWorkspace(const QString &pWorkspace, const QString &pDirName)
+{
+   // Clone the workspace
+
+   git_libgit2_init();
+
+   git_repository *gitRepository = 0;
+   QByteArray workspaceByteArray = pWorkspace.toUtf8();
+   QByteArray dirNameByteArray = pDirName.toUtf8();
+
+   int res = git_clone(&gitRepository, workspaceByteArray.constData(),
+                       dirNameByteArray.constData(), 0);
+
+   if (res) {
+       const git_error *gitError = giterr_last();
+       emit showWarning(tr("Clone Workspace"),
+                        gitError?
+                            tr("Error %1: %2.").arg(QString::number(gitError->klass),
+                                                    Core::formatMessage(gitError->message)):
+                            tr("An error occurred while trying to clone the workspace."));
+   } else if (gitRepository) {
+       git_repository_free(gitRepository);
+   }
+
+   git_libgit2_shutdown();
+}
+
+//==============================================================================
+
+void PmrWebService::doShowExposureFiles(const QString &pExposureUrl)
+{
+    // Show the exposure files, but only if there are some
+
+    if (!mExposureFileNames.values(pExposureUrl).isEmpty()) {
```

Comment:
Remove the curly brackets since both the *if* and *else* statements consist of only one line.